secboot/secboot_hooks.go

```diff
@@ -101,14 +100,19 @@ func SealKeysWithFDESetupHook(runHook fde.RunSetupHookFunc, keys []SealKeyReques
 		if err := skr.BootstrappedContainer.AddKey(skr.SlotName, unlockKey); err != nil {
 			return err
 		}
-		writer := sb.NewFileKeyDataWriter(skr.KeyFile)
-		if err := protectedKey.WriteAtomic(writer); err != nil {
-			return err
-		}
-	}
-	if primaryKey != nil && params.AuxKeyFile != "" {
-		if err := osutil.AtomicWriteFile(params.AuxKeyFile, primaryKey, 0600, 0); err != nil {
-			return fmt.Errorf("cannot write the policy auth key file: %v", err)
+		if skr.KeyFile != "" {
+			writer := sb.NewFileKeyDataWriter(skr.KeyFile)
+			if err := protectedKey.WriteAtomic(writer); err != nil {
```

Comment:
the write atomic is the same on both paths? should it go outside? also maybe this logic could be incapsulated as a method in skr ?